dadjokes.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/rs/zerolog/log"
+)
+
+func dadjoke() string {
+	return httpRequest()
+}
+
+func httpRequest() string {
+
+	// instantiate Zerolog sublogger
+	sublogger := log.With().
+		Str("trigger", "dadjoke").
+		Logger()
+
+	// define dadjoke url and userAgent
+	url := "https://icanhazdadjoke.com/"
+	userAgent := "bytebot-party-pack, an IRC microservice bot funpack thingy. https://github.com/bytebot-chat/bytebot-party-pack/"
+
+	req, err := http.NewRequest("GET", url, nil) // crafting HTTP request with url and userAgent
+	if err != nil {                              // if err, loggging error via Zerolog and returning error to chat
+		sublogger.Warn().Err(err).Msg("HTTP request formatting error")
```

Comment:
We already know it's a an error because you raised an event with a log level of error and included the error messageZ Be more opinionated about the error by explaining it in the Msg portion